ddsrecorder_participants/src/cpp/recorder/mcap/McapHandler.cpp

```diff
@@ -948,71 +1010,98 @@ void McapHandler::rewrite_schemas_nts_()
     schemas_ = new_schemas;
 }
 
-void McapHandler::store_dynamic_types_()
+void McapHandler::generate_dynamic_type_(
```

Comment:
This function is not generating the dynamic type, it's generating it and storing it. I would consider collapsing `generate_dynamic_type` and `store_dynamic_type` into a single `save_dynamic_type` function that, from `type_name`, gets a `type_object`, a `type_information`, generates the `DynamicType` and saves it in the `dynamic_types_` vector.